common/tools/dev-tool/src/config/rollup.base.config.ts

```diff
@@ -129,7 +131,22 @@ export function makeBrowserTestConfig(): RollupOptions {
         }
       }),
       json(),
-      sourcemaps()
+      sourcemaps(),
+      shim({
+        dotenv: `export function config() { }`,
+        fs: `
+          export function stat() { }
+          export function createReadStream() { }
+          export function createWriteStream() { }
+        `,
+        os: `
+          export const type = 1;
+          export const release = 1;
+        `,
+        util: `
+          export function promisify() { }
+        `
+      }),
```

Comment:
This change will make our customers also need to mirror it in their bundling configuration. Can you instead check and see if the browser bundles do not contain these packages in the first place? we usually do that by providing browser -specific drop-in replacements for any node module that use any of these things. For example, karma loads the environment variables in browser tests so TA tests has [an empty browser module for loading variables](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/textanalytics/ai-text-analytics/test/public/utils/env.browser.ts) while [the nodejs one](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/textanalytics/ai-text-analytics/test/public/utils/env.ts) still uses dotenv. The following package.json bit instructs the bundler swap the browser module in: https://github.com/Azure/azure-sdk-for-js/blob/180b4c0b959cc197f3e2a5802d6816c834b6ed16/sdk/textanalytics/ai-text-analytics/package.json#L57-L59

Reply:
Noted. Will revert this and use that drop-in replacement trick in the `recorder-new` package so that all packages using the new unified recorder can make use of it.